src/dialogs/stat_dialogs/barchart/barchart_model.pas

```diff
@@ -0,0 +1,104 @@
+unit barchart_model;
+
+{$mode objfpc}{$H+}
+
+interface
+
+uses
+  Classes, SysUtils, executor, epidatafiles, stat_dialog_contribution,
+  epidatafilestypes,epivaluelabels, outputcreator, freq, epicustombase;
```

Comment:
I'd like to split out the graph functions for sure. This is about 240 lines, including declarations.
I'll try to figure out if we can use a user-defined data source, rather than population directly, but that will be more  confusing, I think.
The rest, about 970 lines, is comparable to means or tables. In general, should all of the statistics commands have separate units estimation, output and result variables.